src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats;
+
+namespace Lib.P2P.Tests
+{
+    [TestClass]
+    public class SwarmTest
+    {
+        private readonly MultiAddress mars =
+            "/ip4/10.1.10.10/tcp/29087/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3";
+
+        private readonly MultiAddress venus =
+            "/ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64";
+
+        private readonly MultiAddress earth =
+            "/ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd";
+
+        private Peer self = new Peer
+        {
+            AgentVersion = "self",
+            Id = "QmXK9VBxaXFuuT29AaPUTgW3jBWZ9JgLVZYdMYTHC6LLAH",
+            PublicKey =
+                "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQCC5r4nQBtnd9qgjnG8fBN5+gnqIeWEIcUFUdCG4su/vrbQ1py8XGKNUBuDjkyTv25Gd3hlrtNJV3eOKZVSL8ePAgMBAAE="
+        };
+
+        private Peer other = new Peer
+        {
+            AgentVersion = "other",
+            Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+            PublicKey =
+                "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+        };
+
+        [TestMethod]
+        public async Task Start_Stop()
+        {
+            var swarm = new SwarmService
+            {
+                LocalPeer = self
+            };
+            await swarm.StartAsync();
+            await swarm.StopAsync();
+        }
+
+        [TestMethod]
+        public void Start_NoLocalPeer()
+        {
+            var swarm = new SwarmService();
+            ExceptionAssert.Throws<NotSupportedException>(() =>
+            {
+                swarm.StartAsync().Wait();
+            });
+        }
+
+        [TestMethod]
+        public void NewPeerAddress()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.RegisterPeerAddress(mars);
+            Assert.IsTrue(swarm.KnownPeerAddresses.Contains(mars));
+        }
+
+        [TestMethod]
+        public void NewPeerAddress_Self()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var selfAddress = "/ip4/178.62.158.247/tcp/4001/ipfs/" + self.Id;
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(selfAddress);
+            });
+
+            selfAddress = "/ip4/178.62.158.247/tcp/4001/p2p/" + self.Id;
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(selfAddress);
+            });
+        }
+
+        [TestMethod]
+        public void NewPeerAddress_BlackList()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.BlackList.Add(mars);
+
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(mars);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)